### PR TITLE
(CLOUD-103) Allow mapping public ip addresses for instances launched in subnets

### PR DESCRIPTION
**Pull request by garethr**

### CODE REVIEW

File: examples/vpc-example/destroy.pp [hunk 1]
```diff
@@ -1,3 +1,8 @@
+ec2_securitygroup { 'sample-sg':
+  ensure => absent,
+  region => 'sa-east-1',
+} ~>
+
```
Comment (cmurphy) [inline examples/vpc-example/destroy.pp]: This belongs in a separate commit.

**Comment by cmurphy** [comment]
This parameter can't be changed after it's set. Puppet will report it as changing on every run without actually doing anything. It should be a read_only() method same as the others.

**Comment by Iristyle** [comment]
Yes, agree with @cmurphy on this one - let's get those issues cleaned up and this should be pretty good to go.

**Comment by garethr** [comment]
@cmurphy cheers. marked read only and moved the security group example out.
